conda/core/link.py

```diff
@@ -218,8 +239,21 @@ def make_unlink_actions(transaction_context, target_prefix, linked_package_data)
     directory_remove_actions = tuple(UnlinkPathAction(transaction_context, linked_package_data,
                                                       target_prefix, d, LinkType.directory)
                                      for d in all_directories)
+
+    if (hasattr(linked_package_data, "preferred_env") and
```

Comment:
`linked_package_data` should always have `preferred_env`.  I'll add a comment on `Record` that'll remove the need for the `hasattr` check